patches/minecraft/net/minecraft/data/tags/TagsProvider.java.patch

```diff
@@ -33,17 +33,25 @@
     }
  
     public String m_6055_() {
-@@ -49,12 +_,13 @@
-          List<TagEntry> list = p_236450_.m_215904_();
-          List<TagEntry> list1 = list.stream().filter((p_236444_) -> {
+@@ -46,15 +_,18 @@
+       this.f_126543_.clear();
+       this.m_6577_();
+       this.f_126543_.forEach((p_236449_, p_236450_) -> {
+-         List<TagEntry> list = p_236450_.m_215904_();
+-         List<TagEntry> list1 = list.stream().filter((p_236444_) -> {
++         // FORGE: Build TagFile directly 
++         TagFile file = p_236450_.buildTagFile();
++         List<TagEntry> list1 = Stream.concat(file.f_215959_().stream(), file.removeEntries().stream()).filter((p_236444_) -> {
```

Comment:
I believe the previous behaviour was to treat removal entries as required, so that's what I did here.

Reply:
Update after doing some searching with GitHub's awful repo search: it does not seem that remove()-d tags were treated as required originally - should I change this back to bring that in line?

Reply:
My intuitive understanding of how tag element removal should work is that they won't error if the element they're trying to remove doesn't exist; it should just be silently ignored.

Reply:
To clarify, the way the system is implemented currently, any tags in the remove list are implicitly treated as optional when deserializing - it is only when building the list that they are treated as required, and only if they are set to required. You can call `TagBuilder.remove(TagEntry.optionalElement(resourceLocation))` to specify that a removed tag isn't required, and it will pass the validation successfully.

Reply:
Update on this: entries listed in `remove` are now treated the same way as entries listed in `values`, meaning the vanilla way of marking a tag as optional should be used to mark an optional entry to remove.